[ApplicationManager] crashes when launching ads

**Description:** [ApplicationManager] crashes when launching ads

**Error:** 

```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.zeoowl.lwp.aquarium, PID: 18862
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.zeoowl.lwp.aquarium/com.google.android.gms.ads.AdActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object java.lang.ref.WeakReference.get()' on a null object reference
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4111)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4277)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2443)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loopOnce(Looper.java:226)
        at android.os.Looper.loop(Looper.java:313)
        at android.app.ActivityThread.main(ActivityThread.java:8751)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object java.lang.ref.WeakReference.get()' on a null object reference
        at com.zeoflow.flowly.ApplicationManager.onActivityPreCreated(ApplicationManager.java:112)
        at android.app.Application.dispatchActivityPreCreated(Application.java:360)
        at android.app.Activity.dispatchActivityPreCreated(Activity.java:1363)
        at android.app.Activity.performCreate(Activity.java:8279)
        at android.app.Activity.performCreate(Activity.java:8270)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1329)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4085)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4277) 
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2443) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loopOnce(Looper.java:226) 
        at android.os.Looper.loop(Looper.java:313) 
        at android.app.ActivityThread.main(ActivityThread.java:8751) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135) 
```

###### To help us triage faster, please check to make sure you are using the [latest version](https://github.com/zeoflow/flowly/releases) of the library.
###### We also happily accept [pull requests](https://github.com/zeoflow/flowly/pulls).

--- a/flowly/src/main/java/com/zeoflow/flowly/ApplicationManager.java
+++ b/flowly/src/main/java/com/zeoflow/flowly/ApplicationManager.java
@@ -109,7 +109,7 @@ public class ApplicationManager implements Application.ActivityLifecycleCallback
     public void onActivityPreCreated(@NonNull Activity activity, @Nullable Bundle savedInstanceState) {
         if (lastActivity == null) {
             lastActivity = new WeakReference<>(activity);
-        } else if (currentActivity.get() != null && currentActivity != lastActivity) {
+        } else if (currentActivity != lastActivity && currentActivity != null) {
             lastActivity = new WeakReference<>(currentActivity.get());
         }
         currentActivity = new WeakReference<>(activity);

--- a/gradle.properties
+++ b/gradle.properties
@@ -23,7 +23,7 @@ android.nonTransitiveRClass=true
 # Maven
 POM_NAME=Flowly
 POM_PACKAGING=aar
-VERSION_NAME=1.0.0
+VERSION_NAME=1.0.1
 GROUP=com.zeoflow.flowly
 POM_DESCRIPTION=Helps you manage your app states easier.
 POM_URL=https://github.com/zeoflow/flowly

